rmpv/src/ext/se.rs

```diff
@@ -253,6 +264,289 @@ impl ser::Serializer for Serializer {
     }
 }
 
+pub struct ExtSerializer {
+    tag: Option<i8>,
+    binary: Option<Vec<u8>>,
+}
+
+impl ser::Serializer for &mut ExtSerializer {
+    type Ok = ();
+    type Error = Error;
+
+    type SerializeSeq = Self;
+    type SerializeTuple = Self;
+    type SerializeTupleStruct = Self;
+    type SerializeTupleVariant = Self;
+    type SerializeMap = Self;
+    type SerializeStruct = Self;
+    type SerializeStructVariant = Self;
+
+    #[inline]
+    fn serialize_i8(self, value: i8) -> Result<Self::Ok, Self::Error> {
+        if self.tag.is_none() {
+            self.tag.replace(value);
+            Ok(())
+        } else {
+            Err(<Error as ser::Error>::custom("Expected bytes"))
+        }
+    }
+
+    #[inline]
+    fn serialize_bytes(self, val: &[u8]) -> Result<Self::Ok, Self::Error> {
+        if self.tag.is_some() {
+            if self.binary.is_none() {
+                self.binary.replace(val.to_vec());
+
+                Ok(())
+            } else {
+                Err(<Error as ser::Error>::custom("Unexpected bytes"))
+            }
+        } else {
+            Err(<Error as ser::Error>::custom("Expected i8 tag"))
+        }
+    }
+
+
+    #[inline]
+    fn serialize_bool(self, _val: bool) -> Result<Self::Ok, Self::Error> {
+        Err(<Error as ser::Error>::custom("Invalid type"))
```

Comment:
Could you mention what is expected in these error messages as well?

It's fine if they're all the same. Having something like `"expected ext tag then bytes, found invalid type"` rather than  `"Invalid type"` would be better, though. As per `custom` documentation, the message should not be capitalized. Then it should be much easier for people to debug these errors if they mention the ext type.